dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java

```diff
@@ -495,8 +495,18 @@ public void printReplacedSql(String content, String formatSql, String rgex, Map<
         //parameter print style
         logger.info("after replace sql , preparing : {}", formatSql);
         StringBuilder logPrint = new StringBuilder("replaced sql , parameters:");
-        for (int i = 1; i <= sqlParamsMap.size(); i++) {
-            logPrint.append(sqlParamsMap.get(i).getValue() + "(" + sqlParamsMap.get(i).getType() + ")");
+        if(sqlParamsMap == null) { 
+        	logger.info("printReplacedSql: sqlParamsMap is null.");
+        }
+        else {         	
+        	for (int i = 1; i <= sqlParamsMap.size(); i++) {
+        		if(sqlParamsMap.get(i) == null ) { 
+        			logger.info("printReplacedSql:  sqlParamsMap.get({}) is null.", i);
```

Comment:
I just want to confirm something, it seems that this method is used to print some parameters for debug?
You may need to modify the print log content here, because I may not know what is `sqlParamsMap`, a more friendly log should tell me which parameter is null. 
It seems there is no special meaning to print` i` here, because I can't know which parameter is null.  The `i` cannot represent the position of the parameter in SQL, am I right?

Reply:
Hi, 
For the friendly logging  of a parameter that is null, i should return 'paramName' that wasn't found in `paramsMap` if i understand correctly. To do so i should probably check if the sql parameter is null in the `setSqlParamsMap`, as i have that information while setting the `sqlParams`. 

Should i still `sqlParamsMap.put(index, prop)` (in `setSqlParamsMap` method) if the param sql is null. I think i shouldn't put in the first place a null `Property` in the sqlParamsMap. And as you said the `i` doesn't really represent anything as the 'Matcher' (that was responsible for setting the `i` ) doesn't order its matches according to its position in the String, correct me if i'm wrong. 

I'll work on these points. 

I apologize if my message wasn't well formed,  still new to commenting.